ssl/statem/statem_clnt.c

```diff
@@ -2558,12 +2559,12 @@ MSG_PROCESS_RETURN tls_process_new_session_ticket(SSL *s, PACKET *pkt)
     RAW_EXTENSION *exts = NULL;
     PACKET nonce;
 
+    PACKET_null_init(&nonce);
+
     if (!PACKET_get_net_4(pkt, &ticket_lifetime_hint)
         || (SSL_IS_TLS13(s)
             && (!PACKET_get_net_4(pkt, &age_add)
-                || !PACKET_get_length_prefixed_1(pkt, &nonce)
-                || !PACKET_memdup(&nonce, &s->session->ext.tick_nonce,
-                                  &s->session->ext.tick_nonce_len)))
+                || !PACKET_get_length_prefixed_1(pkt, &nonce)))
         || !PACKET_get_net_2(pkt, &ticklen)
         || (!SSL_IS_TLS13(s) && PACKET_remaining(pkt) != ticklen)
         || (SSL_IS_TLS13(s)
```

Comment:
On **side note** I find this less readable. I mean (!X && Y) || (X && Z) thing. X ? Z : Y would be much more readable, e.g.
```
        || (SSL_IS_TLS13(s) ? (ticklen == 0 || PACKET_remaining(pkt) < ticklen)
                            : (PACKET_remaining(pkt) != ticklen))) {
```
On related **side note** one can also wonder if it would be appropriate to split the conditions for better troubleshoot-ability. By this I mean following. As of now it will say there was fatal condition on line X without specifying which triggered it. And question is if in case of trouble it would useful to get line number that varies with actual error. For example at least if one could tell apart TLS 1.3 and pre-1.3... Once again, side notes...

Reply:
I updated this using your X ? Z : Y suggestion.